pep-0011.txt

```diff
@@ -14,109 +14,116 @@ Post-History: 18-Aug-2007,
-not be accepted without the platform having official support.
-
-CPU architecture and compiler support are viewed in a similar manner
-as platforms. For example, to consider the ARM architecture supported
-a buildbot running on ARM would be required along with support from
-the Python development team. In general it is not required to have
-a CPU architecture run under every possible platform in order to be
-considered supported.
+This PEP also lists what plaforms *are* supported by the CPython
+interpreter. This lets people know what platforms are directly
+supported by the CPython development team.
 
-Unsupporting platforms
-----------------------
 
-If a certain platform that currently has special code in CPython is
-deemed to be without enough Python users or lacks proper support from
-the Python development team and/or a buildbot, a note must be posted
-in this PEP that this platform is no longer actively supported.  This
-note must include:
+Support tiers
+=============
 
-- the name of the system
-- the first release number that does not support this platform
-  anymore, and
-- the first release where the historical support code is actively
-  removed
+Platform support is broken down into *tiers*. Each tier comes with
+different requirements which lead to different promises being made
+about support.
 
-In some cases, it is not possible to identify the specific list of
-systems for which some code is used (e.g. when autoconf tests for
-absence of some feature which is considered present on all
-supported systems).  In this case, the name will give the precise
-condition (usually a preprocessor symbol) that will become
-unsupported.
+Tier 1
+------
 
-At the same time, the CPython source code must be changed to
-produce a build-time error if somebody tries to install Python on
-this platform.  On platforms using autoconf, configure must fail.
-This gives potential users of the platform a chance to step
-forward and offer maintenance.
+- `CI failures <https://github.com/python/cpython/actions/workflows/build.yml?query=branch%3Amain+is%3Acompleted>`__ block releases.
+- Changes which would break the ``main`` branch are not allowed to be merged;
+  any breakage may be reverted immediately.
+- All core developers are responsible to keep these platforms,
+  and thus ``main``, working.
+- Promotion to this tier requires team consensus/SC approval.
+
+====================== =====
+Target Triple          Notes
+====================== =====
+i686-pc-windows-msvc
+x86_64-pc-windows-msvc
+x86_64-apple-darwin    BSD libc, clang
+x86_64-\*-linux-gnu    glibc, gcc
+====================== =====
+
+
+Tier 2
+------
+
+- Must have a reliable buildbot.
+- At least **two** core developers are signed up to support the platform.
+- Changes which break any of these platforms are to be fixed or
+  reverted within 24 hours.
+- Failures on these platforms block a release.
+- Promotion to this tier requires consensus/SC approval.
+
```

Comment:
I'm not sure that buildbot URLs are reliable. In the past, some ids changed. Moreover, an URL is not easy to read.

Can you either replace the URL with the buildbot name, or add a label to the URL?

For example, replace ``https://buildbot.python.org/all/#/builders/125`` with ``aarch64 Fedora Stable 3.x``.

Reply:
Or better yet (and per PEP 12),

```rst
`aarch64 Fedora Stable 3.x <https://buildbot.python.org/all/#/builders/125>`__
```

Reply:
@CAM-Gerlach that suggestion doesn't work since Victor's worry is about URL stability.

@vstinner what I can do is once this update goes in have a `tier-2` label added to the appropriate buildbots and then simply link to that label.

Reply:
I understand that; I thought by "add a label to the URL" he meant using a link with the buildbot name as descriptive link text, `aarch64 Fedora Stable 3.x`, so it could still be located if its URL changes, but I'm guessing its referring to something else with the buildbots I'm unfamiliar with?

Reply:
He's referring to the labels you can apply to individual buildbots and then pull up a list of them, e.g. all of the 3.10 buildbots can be found with the `3.10` label at https://buildbot.python.org/all/#/builders?tags=%2B3.10 .

Reply:
Gotcha, thanks :+1: